tests/components/influxdb/test_sensor.py

```diff
@@ -0,0 +1,537 @@
+    Mocking the client library there prevents failures and mocking the queue
+    to return `None` on get makes the listener shutdown immediately after initialization.
+    """
+    with patch(f"{INFLUXDB_PATH}.InfluxDBClient") as mock_v1_client, patch(
+        f"{INFLUXDB_PATH}.InfluxDBClientV2"
+    ) as mock_v2_client, patch(
+        f"{INFLUXDB_PATH}.queue.Queue.get", return_value=None
+    ) as queue_get:
+        yield (mock_v1_client, mock_v2_client, queue_get)
+
+
+@pytest.fixture(autouse=True, scope="module")
+def mock_client_close():
+    """Mock close method of clients at module scope."""
+    with patch(f"{INFLUXDB_CLIENT_PATH}.close") as close_v1, patch(
+        f"{INFLUXDB_CLIENT_PATH}V2.close"
+    ) as close_v2:
+        yield (close_v1, close_v2)
+
+
+def _make_v1_resultset(*args):
+    """Create a mock V1 resultset."""
+    for arg in args:
+        yield {"value": arg}
+
+
+def _make_v2_resultset(*args):
+    """Create a mock V2 resultset."""
+    tables = []
+
+    for arg in args:
+        values = {"_value": arg}
+        record = Record(values)
+        tables.append(Table([record]))
+
+    return tables
+
+
+def _set_query_mock_v1(mock_influx_client, return_value=None, side_effect=None):
+    """Set return value or side effect for the V1 client."""
+    query_api = mock_influx_client.return_value.query
+    if side_effect:
+        query_api.side_effect = side_effect
+
+    else:
+        if return_value is None:
+            return_value = []
+
+        def get_return_value(query, **kwargs):
+            """Return mock for test query, return value otherwise."""
+            if query == TEST_QUERY_V1:
+                return MagicMock()
+
+            query_output = MagicMock()
+            query_output.get_points.return_value = return_value
+            return query_output
+
+        query_api.side_effect = get_return_value
+
+    return query_api
+
+
+def _set_query_mock_v2(mock_influx_client, return_value=None, side_effect=None):
+    """Set return value or side effect for the V2 client."""
+    query_api = mock_influx_client.return_value.query_api.return_value.query
+    if side_effect:
+        query_api.side_effect = side_effect
+    else:
+        if return_value is None:
+            return_value = []
+
+        query_api.return_value = return_value
+
+    return query_api
+
+
+async def _setup(hass, config_ext, queries, expected_sensors):
+    """Create client and test expected sensors."""
+    config = {
+        DOMAIN: {},
+        sensor.DOMAIN: {"platform": DOMAIN},
+    }
+    influx_config = config[sensor.DOMAIN]
+    influx_config.update(config_ext)
+    influx_config.update(queries)
+
+    assert await async_setup_component(hass, sensor.DOMAIN, config)
+    await hass.async_block_till_done()
+
+    sensors = []
+    for expected_sensor in expected_sensors:
+        state = hass.states.get(expected_sensor)
+        assert state is not None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)